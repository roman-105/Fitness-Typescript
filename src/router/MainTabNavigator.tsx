import React, { useEffect, useCallback } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { useNavigation } from '@react-navigation/native';
import { bottomBarIcon, bottomBarLabel } from '../components/BottomBarIcon';
import { HeaderContainer } from '../components/Header';
import theme from '../theme';
import { Routes } from './routes';

import HomeScreen from '../screens/home/Home';
import ClubScreen from '../screens/club/Club';
import ExploreScreen from '../screens/explore/Explore';
import ProgressScreen from '../screens/progress/Progress';
import CoachScreen from '../screens/coach/Coach';
import usePrivate from '../hooks/usePrivate';

const Tab = createBottomTabNavigator();

const headerConfigLookup: any = {
  [Routes.Home]: (navigateTo: (route: Routes) => void) => ({
    route: Routes.Home,
    left: [{ icon: 'profile', onPress: () => navigateTo(Routes.Profile) }],
    right: [
      { icon: 'card', onPress: () => navigateTo(Routes.QR) },
      { icon: 'notifications', onPress: () => navigateTo(Routes.Notifications) }
    ]
  }),
  [Routes.Programs]: (navigateTo: (route: Routes) => void) => ({
    route: Routes.Programs,
    left: [{ icon: 'profile', onPress: () => navigateTo(Routes.Profile) }],
    right: [
      { icon: 'search', onPress: () => navigateTo(Routes.Search) },
      { icon: 'notifications', onPress: () => navigateTo(Routes.Notifications) }
    ]
  }),
  [Routes.Progress]: (navigateTo: (route: Routes) => void) => ({
    route: Routes.Progress,
    left: [{ icon: 'profile', onPress: () => navigateTo(Routes.Profile) }],
    right: [
      { icon: 'search', onPress: () => navigateTo(Routes.Search) },
      { icon: 'notifications', onPress: () => navigateTo(Routes.Notifications) }
    ]
  }),
  [Routes.Coach]: (navigateTo: (route: Routes) => void) => ({
    route: Routes.Coach,
    left: [{ icon: 'profile', onPress: () => navigateTo(Routes.Profile) }],
    right: [
      { icon: 'search', onPress: () => navigateTo(Routes.Search) },
      { icon: 'notifications', onPress: () => navigateTo(Routes.Notifications) }
    ]
  }),
  [Routes.Club]: (navigateTo: (route: Routes) => void) => ({
    route: Routes.Club,
    left: [{ icon: 'profile', onPress: () => navigateTo(Routes.Profile) }],
    right: [
      { icon: 'card', onPress: () => navigateTo(Routes.QR) },
      { icon: 'notifications', onPress: () => navigateTo(Routes.Notifications) }
    ]
  })
};

const MainTabNavigator = ({
  Home = HomeScreen,
  Programs = ExploreScreen,
  Progress = ProgressScreen,
  Coach = CoachScreen,
  Club = ClubScreen
}) => {
  const navigation = useNavigation();
  const indexTab: number | undefined = navigation.dangerouslyGetState().routes[0]?.state?.index;

  const navigateTo = useCallback(
    (route: Routes) => {
      navigation.navigate(route);
    },
    [navigation]
  );

  useEffect(() => {
    const navState: any = navigation.dangerouslyGetState().routes[0]?.state;
    // indexTab === undefined => firstTab

    const headerConfig =
      indexTab !== undefined
        ? headerConfigLookup[navState?.routeNames[indexTab]](navigateTo)
        : headerConfigLookup[Routes.Home](navigateTo);

    navigation.setOptions({
      headerLeft: () => <HeaderContainer route={headerConfig.route} items={headerConfig.left} />,
      headerRight: () => <HeaderContainer route={headerConfig.route} items={headerConfig.right} />
    });
  }, [navigation, indexTab, navigateTo]);

  // make the whole tab navigator private
  usePrivate();

  return (
    <Tab.Navigator
      screenOptions={({ route }) => ({
        tabBarIcon: bottomBarIcon(route.name as Routes),
        tabBarLabel: bottomBarLabel(route.name as Routes)
      })}
      tabBarOptions={{
        style: { height: 48 },
        showLabel: true,
        inactiveTintColor: '#838386', // this color only appears here
        activeTintColor: theme.colors.primary.orange
      }}
    >
      <Tab.Screen options={{ tabBarTestID: 'home-tab' }} name={Routes.Home} component={Home} />
      <Tab.Screen
        options={{ tabBarTestID: 'programs-tab' }}
        name={Routes.Programs}
        component={Programs}
      />
      <Tab.Screen
        options={{ tabBarTestID: 'progress-tab' }}
        name={Routes.Progress}
        component={Progress}
      />
      <Tab.Screen options={{ tabBarTestID: 'coach-tab' }} name={Routes.Coach} component={Coach} />
      <Tab.Screen options={{ tabBarTestID: 'club-tab' }} name={Routes.Club} component={Club} />
    </Tab.Navigator>
  );
};

export default MainTabNavigator;
