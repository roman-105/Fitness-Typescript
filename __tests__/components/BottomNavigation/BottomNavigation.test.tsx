import React from 'react';
import { Text } from 'react-native';
import { fireEvent, render, act } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import { Provider } from 'react-redux';
import * as reactRedux from 'react-redux';
import Router from '../../../src/router/Router';
import MainTabNavigator from '../../../src/router/MainTabNavigator';
import configureStore from 'redux-mock-store';

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      dangerouslyGetState: () => ({ routes: [] }),
      dangerouslyGetParent: () => {},
      navigate: () => {},
      setOptions: () => {},
      reset: () => {}
    })
  };
});

jest.mock('@react-native-community/netinfo', () => {});

jest.mock('react-native-static-safe-area-insets', () => {
  const actualInsets = jest.requireActual('react-native-static-safe-area-insets');
  return {
    ...actualInsets,
    StaticSafeAreaInsets: () => ({
      safeAreaInsetsBottom: () => {},
      safeAreaInsetsTop: () => {}
    })
  };
});

jest.mock('react-native-share', () => {});

const mockStore = configureStore([]);

describe('BottomNavigation', () => {
  const initialStore = {
    screenModel: { screens: {} },
    appModel: { ui: {}, walkThrough: {} },
    authModel: {},
    memberModel: { member: {} }
  };
  let store = mockStore(initialStore);

  beforeEach(() => {
    jest.useFakeTimers();
    store = mockStore(initialStore);
    // @ts-ignore
    jest.spyOn(reactRedux, 'useDispatch').mockImplementation(() => ({
      appModel: {
        checkIfWalkThroughIsDone: jest.fn()
      },
      screenModel: {
        fetch: jest.fn()
      },
      authModel: {
        refreshAuthedAxios: jest.fn()
      },
      memberModel: {
        getMemberInfo: jest.fn()
      },
      modalModel: {
        setInfo: jest.fn()
      }
    }));
  });

  it('renders the correct element', async () => {
    const { getByText } = render(
      <Provider store={store}>
        <NavigationContainer>
          <Router />
        </NavigationContainer>
      </Provider>
    );

    const textElement1 = getByText('Home');
    const textElement2 = getByText('Explore');
    const textElement3 = getByText('Progress');
    const textElement4 = getByText('Coach');
    const textElement5 = getByText('Clubs');

    await act(async () => {
      expect(textElement1).not.toBeNull();
      expect(textElement2).not.toBeNull();
      expect(textElement3).not.toBeNull();
      expect(textElement4).not.toBeNull();
      expect(textElement5).not.toBeNull();
    });
  });

  it('renders the content in the right order', async () => {
    const { toJSON } = render(
      <Provider store={store}>
        <NavigationContainer>
          <MainTabNavigator
            Home={() => <Text>mock1</Text>}
            Programs={() => <Text>mock2</Text>}
            Progress={() => <Text>mock3</Text>}
            Coach={() => <Text>mock4</Text>}
            Club={() => <Text>mock5</Text>}
          />
        </NavigationContainer>
      </Provider>
    );

    await act(async () => expect(toJSON()).toMatchSnapshot());
  });

  it('different color if the item is the active page', async () => {
    const { toJSON, getByText } = render(
      <Provider store={store}>
        <NavigationContainer>
          <MainTabNavigator
            Home={() => <Text>mock1</Text>}
            Programs={() => <Text>mock2</Text>}
            Progress={() => <Text>mock3</Text>}
            Coach={() => <Text>mock4</Text>}
            Club={() => <Text>mock5</Text>}
          />
        </NavigationContainer>
      </Provider>
    );

    const secondTab = getByText('Explore');
    await act(async () => {
      await fireEvent.press(secondTab);
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
