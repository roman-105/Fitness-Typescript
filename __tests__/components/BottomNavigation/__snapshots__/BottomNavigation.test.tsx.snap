// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomNavigation different color if the item is the active page 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        enabled={true}
        pointerEvents="none"
        removeClippedSubviews={true}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            null,
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "top": 30000,
            }
          }
        >
          <View
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <Text>
              mock1
            </Text>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        enabled={true}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            null,
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <Text>
              mock2
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "rgb(255, 255, 255)",
          "borderTopColor": "rgb(216, 216, 216)",
          "borderTopWidth": 0.5,
          "bottom": 0,
          "elevation": 8,
          "height": 48,
          "left": 0,
          "paddingBottom": 0,
          "paddingHorizontal": 0,
          "position": null,
          "right": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
          testID="home-tab"
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#FE7000"
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#838386"
              />
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Argumentum",
                },
                Object {
                  "color": "#838386",
                },
                Object {
                  "fontSize": 11,
                },
                Object {
                  "lineHeight": 12,
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Home
          </Text>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": true,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
          testID="programs-tab"
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#FE7000"
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#838386"
              />
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Argumentum",
                },
                Object {
                  "color": "#FE7000",
                },
                Object {
                  "fontSize": 11,
                },
                Object {
                  "lineHeight": 12,
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Explore
          </Text>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
          testID="progress-tab"
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#FE7000"
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#838386"
              />
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Argumentum",
                },
                Object {
                  "color": "#838386",
                },
                Object {
                  "fontSize": 11,
                },
                Object {
                  "lineHeight": 12,
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Progress
          </Text>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
          testID="coach-tab"
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#FE7000"
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#838386"
              />
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Argumentum",
                },
                Object {
                  "color": "#838386",
                },
                Object {
                  "fontSize": 11,
                },
                Object {
                  "lineHeight": 12,
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Coach
          </Text>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
          testID="club-tab"
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#FE7000"
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#838386"
              />
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Argumentum",
                },
                Object {
                  "color": "#838386",
                },
                Object {
                  "fontSize": 11,
                },
                Object {
                  "lineHeight": 12,
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Clubs
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`BottomNavigation renders the content in the right order 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        enabled={true}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            null,
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <Text>
              mock1
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "rgb(255, 255, 255)",
          "borderTopColor": "rgb(216, 216, 216)",
          "borderTopWidth": 0.5,
          "bottom": 0,
          "elevation": 8,
          "height": 48,
          "left": 0,
          "paddingBottom": 0,
          "paddingHorizontal": 0,
          "position": null,
          "right": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": true,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
          testID="home-tab"
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#FE7000"
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#838386"
              />
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Argumentum",
                },
                Object {
                  "color": "#FE7000",
                },
                Object {
                  "fontSize": 11,
                },
                Object {
                  "lineHeight": 12,
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Home
          </Text>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
          testID="programs-tab"
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#FE7000"
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#838386"
              />
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Argumentum",
                },
                Object {
                  "color": "#838386",
                },
                Object {
                  "fontSize": 11,
                },
                Object {
                  "lineHeight": 12,
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Explore
          </Text>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
          testID="progress-tab"
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#FE7000"
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#838386"
              />
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Argumentum",
                },
                Object {
                  "color": "#838386",
                },
                Object {
                  "fontSize": 11,
                },
                Object {
                  "lineHeight": 12,
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Progress
          </Text>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
          testID="coach-tab"
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#FE7000"
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#838386"
              />
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Argumentum",
                },
                Object {
                  "color": "#838386",
                },
                Object {
                  "fontSize": 11,
                },
                Object {
                  "lineHeight": 12,
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Coach
          </Text>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
          testID="club-tab"
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#FE7000"
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <test-file-stub
                fill="#838386"
              />
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Argumentum",
                },
                Object {
                  "color": "#838386",
                },
                Object {
                  "fontSize": 11,
                },
                Object {
                  "lineHeight": 12,
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Clubs
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
